transform_inline.ml

```diff
@@ -0,0 +1,312 @@
+open Instr
+open Edit
+
+(* Unlike `analysis_input`, this preserves the order of `formals`. This
+   facilitates matching the formals to the arguments passed at a callsite.*)
+type inlining_input = {
+  formals : variable list;
+  instrs : instructions;
+}
+
+let as_inlining_input (func:afunction) (version:version) : inlining_input =
+  { formals = Analysis.as_var_list func.formals; instrs = version.instrs }
+
+(* FUNCTION INLINING *)
+(* Given a program, inline the functions in it to the maximum possible extent.
+   Transitively recursive calls are inlined until they become self recursive.
+   Self recursive callsites are not inlined. Intermediate inlined versions are
+   stored in the caller. *)
+let inline ({main; functions} as orig_prog : program) : program option =
+
+  (* Given the formals and instructions of a function, get all the declared and
+     used vars in the function. *)
+  let function_vars {formals; instrs} =
+    let all_declared_vars instrs = Array.fold_left
```

Comment:
This will be moved to `instr.ml` once we change `analysis_input` to contain a `List` of `formals`.